csm_web/scheduler/models.py

```diff
@@ -232,19 +254,36 @@ class WaitlistedStudent(Profile):
     )
     timestamp = models.DateTimeField(auto_now_add=True)
 
-    def add_from_waitlist(self, section_to_enroll):
+    @classmethod
+    def add_student_from_waitlist(self, section):
         """
-        Removes this student from all waitlists and adds them as a student in the given section.
+        Adds a waitlisted student into the section and removes the waitlisted student from all waitlists for the course.
+        Only does the above is capacity is not reached (e.g. an exception was made to add an additional student to section)
         """
-        WaitlistedStudent.objects.filter(user=self.user, active=True, course=self.section.mentor.course).update(active=False)
-
-        student = Student.objects.create(
-            user=self.user,
-            section=section_to_enroll
-        )
-        if student:
-            logger.info("Student created successfully")
-
+        if section.open_capacity: 
+
+            course_queryset = WaitlistedStudent.objects.filter(active=True, course=section.course)
+            section_waitlist = course_queryset.filter(section=section.id).order_by() # position?
+            
+            if section_waitlist.exists():
+                waitlisted_student = section_waitlist.first()
+                
+                student = Student.objects.create(
+                    user=waitlisted_student.user,
+                    section=section,
+                    course=section.course,
+                )
+                if student:
+                    logger.info("Student added from waitlist successfully")
+                
+                waitlisted_student.active = False  # Mark the waitlisted student as inactive similar to drop student
+                student.save()
+                # TODO: update positions of all other waitlisted students
+
+            # TODO: confusion around whether we set active = False or delete
+            student_waitlist = course_queryset.filter(user=student.user)
+            for waitlist in student_waitlist:
+                waitlist.delete()
```

Comment:
There are some TODOs here that should probably be resolved at some point (no need to do that here if that's not the plan).

Things to think about:
* This can probably be extracted to its own helper function as a pure python function, since it'll be called by multiple places.
* You can specify an ordering within the model itself (with the `ordering` field in an inner `Meta` class), so that you do not need to worry about implementing a sort here.
* The ordering here would be by timestamp, which would work fine, unless we want to allow coordinators to re-order the waitlist. This isn't something to worry about now, but an alternative is to just associate an integer with each student, and preserve the invariant that sorting by these integers gives the order in the waitlist.